src/modules/powerrename/PowerRenameUILib/ExplorerItem.xaml

```diff
@@ -2,69 +2,124 @@
+
+
+            <CheckBox
+                Name="{x:Bind IdStr}"
+                Grid.Column="1"
+                Width="28"
+                Height="28"
+                MinWidth="0"
+                AutomationProperties.HelpText="{x:Bind Renamed}"
+                AutomationProperties.Name="{x:Bind Original}"
+                Content=""
+                IsChecked="{x:Bind Checked, Mode=TwoWay}"
+                IsTabStop="True"
+                TabIndex="0"
+                XYFocusKeyboardNavigation="Enabled" />
+            <Image
+                Grid.Column="2"
+                Width="16"
+                Margin="4,0,0,0"
+                HorizontalAlignment="Left"
+                Source="{x:Bind ImagePath}" />
+            <TextBlock
+                Grid.Column="3"
+                Margin="6,0,4,0"
+                VerticalAlignment="Center"
+                FontSize="12"
+                Foreground="{ThemeResource TextFillColorSecondaryBrush}"
+                Text="{x:Bind Original, Mode=OneWay}"
+                TextTrimming="CharacterEllipsis" />
+        </Grid>
+        <TextBlock
+            Grid.Column="2"
+            Margin="0,0,4,0"
+            VerticalAlignment="Center"
+            FontSize="12"
+            FontWeight="SemiBold"
+            Foreground="{ThemeResource AccentTextFillColorPrimaryBrush}"
+            Text="{x:Bind Renamed, Mode=OneWay}"
+            TextTrimming="CharacterEllipsis" />
+
+        <Border
+            x:Name="ErrorIcon"
+            Grid.Column="3"
+            Width="12"
+            Height="12"
+            CornerRadius="12"
+            Margin="4,0,8,0"
+            Background="{ThemeResource SystemFillColorCriticalBrush}"
+            Visibility="Collapsed">
+            <TextBlock
+                HorizontalAlignment="Center"
+                VerticalAlignment="Center"
+                FontFamily="{ThemeResource SymbolThemeFontFamily}"
+                Foreground="White"
+                FontSize="8"
+                Text="&#xE10A;" />
+            <ToolTipService.ToolTip>
+                <TextBlock
+                    x:Name="ErrorMessageTxt"
+                    Text="Error message goes here"
+                    TextWrapping="Wrap" />
+            </ToolTipService.ToolTip>
+        </Border>
+
+
+
         <VisualStateManager.VisualStateGroups>
             <VisualStateGroup x:Name="CommonStates">
-                <VisualState x:Name="Normal"/>
+                <VisualState x:Name="Normal" />
+                <VisualState x:Name="Highlight">
+                    <VisualState.Setters>
+                        <Setter Target="HighlightBorder.Opacity" Value="0.1" />
+                        <Setter Target="HighlightBorder.BorderBrush" Value="{ThemeResource AccentTextFillColorPrimaryBrush}" />
+                        <Setter Target="HighlightBorder.Background" Value="{ThemeResource AccentTextFillColorPrimaryBrush}" />
+                    </VisualState.Setters>
```

Comment:
Correct me if I'm mistaken, but I don't see a `Icon visibility = none` here. Isn't that needed to hide it when it's back to normal?

Reply:
No, the closing tag indicates that the Normal state falls back to the default states defined in the XAML - in this case, the icon is already collapsed.

Reply:
Error -> Highlight state works as expeected